lib/presentation/shared_widgets/country_search_dialog.dart

```diff
@@ -0,0 +1,88 @@
+import 'package:collaction_app/presentation/authentication/utils/countries.dart';
+import 'package:flutter/material.dart';
+
+class CountrySearch extends StatefulWidget {
+  final Function(Country)? onCountrySelected;
+  const CountrySearch({Key? key, this.onCountrySelected}) : super(key: key);
+
+  @override
+  _CountrySearchState createState() => _CountrySearchState();
+}
+
+class _CountrySearchState extends State<CountrySearch> {
+  final _countries = List<Country>.from(countries);
+
+  @override
+  Widget build(BuildContext context) {
+    return SizedBox(
+      width: double.maxFinite,
+      child: Column(
+        mainAxisSize: MainAxisSize.min,
+        children: [
+          TextField(
+            decoration: const InputDecoration(
+              prefixIcon: Icon(Icons.search),
+            ),
+            onChanged: (text) {
```

Comment:
Can we abstract this to a method to keep the UI code a bit cleaner

Reply:
Sure let me move it to a `private` method